### PR TITLE
Update documentation for ingress annotation validation

### PR DESCRIPTION
**Pull request by mikestephen**
### Proposed changes
Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue here in this description (not in the title of the PR).

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by Dean-Coakley — APPROVED** [review]

File: docs-web/configuration/ingress-resources/advanced-configuration-with-annotations.md [hunk 1]
```diff
@@ -39,6 +39,34 @@ spec:
...
+$ kubectl describe ing cafe-ingress-with-annotations
+. . .
+Events:
+  Type     Reason    Age   From                      Message
+  ----     ------    ----  ----                      -------
+  Warning  Rejected  13s   nginx-ingress-controller  annotations.nginx.org/redirect-to-https: Invalid value: "yes please": must be a boolean
+```
+Note how the events section includes a Warning event with the Rejected reason.
+
+**Note**: If you make an existing Ingress invalid, the Ingress Controller will reject it and remove the corresponding configuration from NGINX.
```
Comment (pleshakov) [inline docs-web/configuration/ingress-resources/advanced-configuration-with-annotations.md:68]: would it make sense to mention which annotations are not validated?  Looks like this PR https://github.com/nginxinc/kubernetes-ingress/pull/1288 addresses the remaining ones, but the snippets annotations will not be validated. Shall we mention that?

**Review by pleshakov — APPROVED** [review]

File: docs-web/configuration/ingress-resources/advanced-configuration-with-annotations.md (hunk 1 above)
Comment (mikestephen) [reply docs-web/configuration/ingress-resources/advanced-configuration-with-annotations.md:68]: Good idea, I've added that now.

File: docs-web/configuration/ingress-resources/advanced-configuration-with-annotations.md [hunk 2]
```diff
@@ -39,6 +39,38 @@ spec:
...
+
+The following Ingress annotations currently have limited or no validation: `nginx.org/proxy-connect-timeout`, `nginx.org/proxy-read-timeout`, `nginx.org/proxy-send-timeout`, `nginx.org/client-max-body-size`, `nginx.org/proxy-buffers`, `nginx.org/proxy-buffer-size`, `nginx.org/proxy-max-temp-file-size`, `nginx.org/upstream-zone-size`, `nginx.org/fail-timeout`, `nginx.org/server-tokens`, `nginx.org/proxy-hide-headers`, `nginx.org/proxy-pass-headers`, `nginx.org/rewrites`, `nginx.com/jwt-key`, `nginx.com/jwt-realm`, `nginx.com/jwt-token`, `nginx.com/jwt-login-url`, `nginx.org/ssl-services`, `nginx.org/grpc-services`, `nginx.org/websocket-services`, `nginx.com/sticky-cookie-services`, `nginx.com/slow-start`, `appprotect.f5.com/app-protect-policy`, `appprotect.f5.com/app-protect-security-log`.
```
Comment (jputrino) [inline docs-web/configuration/ingress-resources/advanced-configuration-with-annotations.md:70]: suggest formatting as a bullet list for easier scan-ability

```suggestion
The following Ingress annotations currently have limited or no validation: 

- `nginx.org/proxy-connect-timeout`, 
- `nginx.org/proxy-read-timeout`, 
- `nginx.org/proxy-send-timeout`, 
- `nginx.org/client-max-body-size`, 
- `nginx.org/proxy-buffers`, 
- `nginx.org/proxy-buffer-size`, 
- `nginx.org/proxy-max-temp-file-size`, 
- `nginx.org/upstream-zone-size`, 
- `nginx.org/fail-timeout`, 
- `nginx.org/server-tokens`, 
- `nginx.org/proxy-hide-headers`, 
- `nginx.org/proxy-pass-headers`, 
- `nginx.org/rewrites`, 
- `nginx.com/jwt-key`, 
- `nginx.com/jwt-realm`, 
- `nginx.com/jwt-token`, 
- `nginx.com/jwt-login-url`, 
- `nginx.org/ssl-services`, 
- `nginx.org/grpc-services`, 
- `nginx.org/websocket-services`, 
- `nginx.com/sticky-cookie-services`, 
- `nginx.com/slow-start`, 
- `appprotect.f5.com/app-protect-policy`, 
- `appprotect.f5.com/app-protect-security-log`.
```

**Review by jputrino — APPROVED** [review]
Looks great!

File: docs-web/configuration/ingress-resources/advanced-configuration-with-annotations.md (hunk 2 above)
Comment (mikestephen) [reply docs-web/configuration/ingress-resources/advanced-configuration-with-annotations.md:70]: Thanks Jodie, great idea

**Pull request merged**